contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java

```diff
@@ -0,0 +1,164 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.contrib.avro;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.avro.AvroRuntimeException;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.lib.io.fs.AbstractFileInputOperator;
+
+/**
+ * <p>
+ * Avro File Input Operator
+ * </p>
+ * A specific implementation of the AbstractFileInputOperator to read Avro
+ * container files.<br>
+ * No need to provide schema,its inferred from the file<br>
+ * Users can add the {@link IdempotentStorageManager.FSIdempotentStorageManager}
+ * to ensure exactly once semantics with a HDFS backed WAL.
+ * 
+ * @displayName AvroFileInputOperator
+ * @category Input
+ * @tags fs, file,avro, input operator
+ */
+@InterfaceStability.Evolving
+public class AvroFileInputOperator extends AbstractFileInputOperator<GenericRecord>
+{
+
+  private transient long offset = 0L;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  private transient DataFileStream<GenericRecord> avroDataStream;
+  private transient GenericRecord record = null;
+
+  public final transient DefaultOutputPort<GenericRecord> output = new DefaultOutputPort<GenericRecord>();
+
+  public final transient DefaultOutputPort<String> completedFilesPort = new DefaultOutputPort<String>();
+
+  public final transient DefaultOutputPort<String> errorRecordsPort = new DefaultOutputPort<String>();
+
+  /**
+   * Returns a input stream given a file path
+   * 
+   * @param path
+   * @return InputStream
+   * @throws IOException
+   */
+  @Override
+  protected InputStream openFile(Path path) throws IOException
+  {
+    InputStream is = super.openFile(path);
+    if (is != null) {
+      try {
+        DatumReader<GenericRecord> datumReader = new GenericDatumReader<GenericRecord>();
+        avroDataStream = new DataFileStream<GenericRecord>(is, datumReader);
+        datumReader.setSchema(avroDataStream.getSchema());
+      } catch (NullPointerException npe) {
+        LOG.error("Schemaless file", npe);
+        throw new NullPointerException();
+      }
+    }
+    return is;
+  }
+
+  @Override
+  protected GenericRecord readEntity() throws IOException
+  {
+    return readRecord();
+  }
+
+  /**
+   * Reads a GenericRecord from the given input stream<br>
+   * Emits the FileName,Offset,Exception on the error port if its connected
+   * 
+   * @return GenericRecord
+   */
+  private GenericRecord readRecord() throws IOException
+  {
+    record = null;
```

Comment:
I don't see "record" variable being used anywhere else other than this method.
Can you make this in local scope instead of class level?